include/sys/signal.h

```diff
@@ -81,14 +81,18 @@ typedef struct ctx ctx_t;
  *
  * \sa sig_post
  * \note Must be called with p::p_lock held. Returns with p::p_lock held.
+ *
+ * \returns error code if can't signal process
  */
-void sig_kill(proc_t *p, signo_t sig);
+int sig_kill(proc_t *p, signo_t sig);
 
 /*! \brief Signal all processes in a process group.
  *
  * \note Must be called with pg::pg_lock held. Returns with pg::pg_lock held.
+ *
+ * \returns error code if can't signal any of the processes
  */
-void sig_pgkill(pgrp_t *pg, signo_t sig);
+int sig_pgkill(pgrp_t *pg, signo_t sig);
```

Comment:
Likewise, I think the use of `sig_pgkill()` should be similar to use of [`kpgsignal()`](http://bxr.su/NetBSD/sys/kern/kern_sig.c#885): permission checks should be carried out elsewhere. We might need another procedure for signaling a process group on behalf of a user, similar to [`killpg1()`](http://bxr.su/NetBSD/sys/kern/kern_sig.c#809).